metrics/network/network-metrics-nuttcp.sh

```diff
@@ -0,0 +1,202 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Description:
+# This metrics test measures the UDP network bandwidth using nuttcp
+# in a interconnection container-client <----> container-server.
+
+set -e
+
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/lib/network-common.bash"
+source "${SCRIPT_PATH}/../lib/common.bash"
+
+# Measurement time (seconds)
+transmit_timeout="${transmit_timeout:-30}"
+# Image name
+image="${IMAGE:-local-nuttcp}"
+# Dockerfile
+dockerfile="${SCRIPT_PATH}/nuttcp_dockerfile/Dockerfile"
+# Name of the server container
+server_name="${server_name:-network-server}"
+# Nuttcp version
+nuttcp_version="${nuttcp_version:-7.3.2}"
+
+function udp_default_buffer_size {
+	# Test UDP Jumbo (9000 byte MTU) packets
+	# Packet header (ICMP+IP) is 28bytes, so maximum payload is 8972 bytes
+	# See the nuttcp documentation for more hints:
+	# https://fasterdata.es.net/performance-testing/network-troubleshooting-tools/nuttcp/
+	local bl=8972
+	local TEST_NAME="${TEST_NAME:-nuttcp test with ${bl} buffer size}"
+	local result="$(udp_bandwidth)"
+	save_results "$TEST_NAME" "$result"
+}
+
+function udp_specific_buffer_size {
+	# Test UDP standard (1500 byte MTU) packets
+	# Even though the packet header (ICMP+IP) is 28 bytes, which would
+	# result in 1472 byte frames, the nuttcp documentation recommends
+	# use of 1200 byte payloads.
+	# See the nuttcp examples.txt for more information:
+	# http://nuttcp.net/nuttcp/5.1.3/examples.txt
+	local bl=1200
+ 	local TEST_NAME="${TEST_NAME:-nuttcp test with ${bl} buffer size}"
+	local result="$(udp_bandwidth)"
+	save_results "$TEST_NAME" "$result"
+}
+
+function udp_bandwidth {
+	# Arguments to run the client/server
+	local server_extra_args="--name=$server_name"
+	local client_extra_args="--rm"
+
+	local server_command="tail -f /dev/null"
+	local server_address=$(start_server "$image" "$server_command" "$server_extra_args")
+
+	local client_command="/root/nuttcp -T${transmit_timeout} -u -Ru -i1 -l${bl} ${server_address}"
+	local server_command="/root/nuttcp -u -S"
+
+	docker exec ${server_name} sh -c "${server_command}"
```

Comment:
As a user, it'd be helpful for me to see what actual nuttcp arguments are used in the calls.  It'd help for quickly rerunning by hand, or slightly modifying but keeping apples to apples....